File: src/main/java/ac/kr/deu/connect/luck/event/entity/Event.java
package ac.kr.deu.connect.luck.event.entity;

import ac.kr.deu.connect.luck.common.entity.BaseEntity;
import ac.kr.deu.connect.luck.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Event extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    @Column(columnDefinition = "LONGTEXT")
    private String content;
    private String zipCode;
    private String streetAddress;
    private String detailAddress;
    private LocalDateTime startAt;
    private LocalDateTime endAt;
    private String imageUrl;
    @ManyToOne
    private User manager;
    @Enumerated(EnumType.STRING)
    private EventStatus status;
}

File: src/main/kotlin/ac/kr/deu/connect/luck/event/entity/Event.kt
package ac.kr.deu.connect.luck.event.entity

import jakarta.persistence.Column
import jakarta.persistence.ElementCollection
import jakarta.persistence.Entity
import jakarta.persistence.EntityListeners
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import java.time.Instant
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener

@Entity
@EntityListeners(AuditingEntityListener::class)
class Event(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,
    var title: String,
    @Column(columnDefinition = "TEXT")
    var thumbnail: String,
    @Column(columnDefinition = "LONGTEXT")
    var content: String,
    @ElementCollection(fetch = FetchType.LAZY)
    var images: List<String> = emptyList(),
    @Enumerated(EnumType.STRING)
    var status: EventStatus,

    var startDate: Instant? = null,
    var endDate: Instant? = null,

    var zipCode: String? = null,
    var address: String? = null,
    var detailAddress: String? = null,
    var trafficInfo: String? = null,


    val managerId: Long,
    @Column(unique = true)
    val provider: EventProvider? = null,

    @CreatedDate
    val createdAt: Instant = Instant.now(),
    @LastModifiedDate
    val updatedAt: Instant = Instant.now(),
)
